From ab7b1a361d51157118e1a61ce6530a59bcef4b61 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Tue, 16 Apr 2024 16:10:09 +0200
Subject: [PATCH 3/6] nss_dp_main: swap dp_exit function call

First unregister nss_dp platform devices then cleanup the HAL.

This is to fix kernel panic by cleaning data that needs to be used by
platform driver unregister functions.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 nss_dp_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -1161,6 +1161,8 @@ int __init nss_dp_init(void)
  */
 void __exit nss_dp_exit(void)
 {
+	platform_driver_unregister(&nss_dp_drv);
+
 	/*
 	 * TODO Move this to soc_ops
 	 */
@@ -1168,8 +1170,6 @@ void __exit nss_dp_exit(void)
 		nss_dp_hal_cleanup();
 		dp_global_ctx.common_init_done = false;
 	}
-
-	platform_driver_unregister(&nss_dp_drv);
 }
 
 module_init(nss_dp_init);
